### PR TITLE
Small QOL Features

### PR DESCRIPTION
**Pull request by bryanterichardson**
Changes:
- Introduces docker to our app. Definitely not a final setup but a good start
- Refactors code to be a bit more pythonic
- Installs `ipython`
- Updates the README to reflect changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,7 @@
 ### 1. Navigate to roastme/ directory:
```
Comment (bryanterichardson) [inline README.md]: forgot to update this. You dont navigate to the roast me directory. Just do it from the root

File: roastme/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FROM python:3.8.6
```
Comment (bryanterichardson) [inline roastme/Dockerfile:1]: FYI, docker basically has its own repo like system of docker containers. By putting this here, it knows to go and download the stable version of this container. This is how we will always have the same python version (even if we didnt actually have python on our machine)

File: roastme/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+ENV PYTHONUNBUFFERED 1
+
+RUN mkdir -p /usr/src/app
+WORKDIR /usr/src/app
+COPY requirements.txt /usr/src/app
+RUN pip install -r requirements.txt
```
Comment (bryanterichardson) [inline roastme/Dockerfile:7]: this is what ensures all the modules are always installed

File: roastme/main.py [hunk 4]
```diff
@@ -1,28 +1,33 @@
...
-import os
+
 
 TOKEN = os.environ['DISCORD_ROASTME_TOKEN']
+bot = commands.Bot(command_prefix="/")
+
 
-def createInsult(user):
-    includeAdverb = bool(randrange(0,2))
+def create_insult(user):
```
Comment (bryanterichardson) [inline roastme/main.py:13]: use snake case in python FAM

File: roastme/main.py [hunk 5]
```diff
@@ -1,28 +1,33 @@
...
         return user + " is " + adverb + " " + adjective
     else:
         return user + " is " + adjective
-   
-bot = commands.Bot(command_prefix="/")
+
 
 @bot.command()
-async def roast(ctx, arg):
+async def roast(context, arg):
```
Comment (bryanterichardson) [inline roastme/main.py:24]: I dont like using `arg` but ill leave it for now..

File: roastme/main.py [hunk 6]
```diff
@@ -1,28 +1,33 @@
...
     print(arg)
     if (arg == 'me'):
-        await ctx.send("You're garbage.")
+        await context.send("You're garbage.")
     if (arg.startswith("<@", 0)):
```
Comment (bryanterichardson) [inline roastme/main.py:28]: should single quote these but ill give you a pass

File: README.md [hunk 7]
```diff
@@ -1,8 +1,7 @@
...
-### 2. Activate virtual environment:
-~ source bot-env/bin/activate
-### 3. Run bot locally
-~ python3 main.py
+- Run `./bin/dev/setup` from root dir
+### 2. Access the container
+- To run a python shell `docker exec -it roastme_main_1 python`
+- To start the app `docker exec -it roastme_main_1 python main.py`
+- To just access the container `docker exec -it roastme_main_1 bash`
+- To use ipython (recommended) `docker exec -it roastme_main_1 python`
```
Comment (bryanterichardson) [inline README.md]: typo

File: roastme/main.py (hunk 6 above)
Comment (dmartinezgamboa) [reply roastme/main.py:28]: I think my prettier extension does this automatically...

Comment (dmartinezgamboa) [reply roastme/main.py:28]: wait im trolling my prettier wasnt setup for this repo lol. fasho single quotes it is

**Review by dmartinezgamboa — COMMENTED** [review]
Looks good.

**Review by dmartinezgamboa — APPROVED** [review]
Thanks for the tips!

**Pull request merged**